AstarteDeviceSDKCSharpE2E.Tests/AstarteMockData/AstarteMockDevice.cs

```diff
@@ -0,0 +1,279 @@
+            BinaryblobarrayEndpoint = binaryblobarrayEndpoint;
+            DatetimearrayEndpoint = datetimearrayEndpoint;
+        }
+
+        public bool Equals(MockDataDevice other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (other.DatetimeEndpoint == new DateTime() || other.DatetimearrayEndpoint.Length == 0)
+            {
+                return
+            other.IntegerEndpoint == this.IntegerEndpoint &&
+            other.DoubleEndpoint == this.DoubleEndpoint &&
+            other.LongintegerEndpoint == this.LongintegerEndpoint &&
+            other.StringEndpoint == this.StringEndpoint &&
+            other.BooleanEndpoint == this.BooleanEndpoint &&
+            other.BinaryblobEndpoint.SequenceEqual(this.BinaryblobEndpoint) &&
+            other.IntegerarrayEndpoint.SequenceEqual(this.IntegerarrayEndpoint) &&
+            other.DoublearrayEndpoint.SequenceEqual(this.DoublearrayEndpoint) &&
+            other.LongintegerarrayEndpoint.SequenceEqual(this.LongintegerarrayEndpoint) &&
+            other.StringarrayEndpoint.SequenceEqual(this.StringarrayEndpoint) &&
+            ComperByteArrays(other.BinaryblobarrayEndpoint) &&
+            other.BooleanarrayEndpoint.SequenceEqual(BooleanarrayEndpoint);
+            }
+
+            return
+            other.IntegerEndpoint == this.IntegerEndpoint &&
+            other.DoubleEndpoint == this.DoubleEndpoint &&
+            other.LongintegerEndpoint == this.LongintegerEndpoint &&
+            other.StringEndpoint == this.StringEndpoint &&
+            other.BooleanEndpoint == this.BooleanEndpoint &&
+            DateTime.Compare(other.DatetimeEndpoint, this.DatetimeEndpoint) == 0 ? true : false &&
+            other.BinaryblobEndpoint.SequenceEqual(this.BinaryblobEndpoint) &&
+            other.IntegerarrayEndpoint.SequenceEqual(this.IntegerarrayEndpoint) &&
+            other.DoublearrayEndpoint.SequenceEqual(this.DoublearrayEndpoint) &&
+            other.LongintegerarrayEndpoint.SequenceEqual(this.LongintegerarrayEndpoint) &&
+            other.StringarrayEndpoint.SequenceEqual(this.StringarrayEndpoint) &&
+            ComperByteArrays(other.BinaryblobarrayEndpoint) &&
+            other.DatetimearrayEndpoint.SequenceEqual(DatetimearrayEndpoint) &&
+            other.BooleanarrayEndpoint.SequenceEqual(BooleanarrayEndpoint);
+
+        }
+
+        public override bool Equals(object obj) => Equals(obj as MockDataDevice);
+
+        public override int GetHashCode() =>
+        (DoubleEndpoint, DoublearrayEndpoint,
+         IntegerEndpoint, IntegerarrayEndpoint,
+         BooleanEndpoint, BooleanarrayEndpoint,
+         LongintegerEndpoint, LongintegerarrayEndpoint,
+         StringEndpoint, StringarrayEndpoint,
+         DatetimeEndpoint, DatetimearrayEndpoint,
+         BinaryblobEndpoint, BinaryblobarrayEndpoint)
+        .GetHashCode();
+
+        private bool ComperByteArrays(byte[][] arraByte)
```

Comment:
Fix the typo in `ComperByteArrays` and `arraByte`.

Reply:
@osmanhadzic Add these changes.